### PR TITLE
Add optional SyncBatchNorm support to ImageNet example

### PR DESCRIPTION
**Pull request by snehasonkusare-tech**
This PR adds optional SyncBatchNorm support to the ImageNet training script.

### What’s included:
- Added new CLI argument: `--sync-bn`
- If both (a) distributed training is enabled and (b) `--sync-bn` is passed,
  BatchNorm layers are converted to SyncBatchNorm using:

    nn.SyncBatchNorm.convert_sync_batchnorm(model)

- Conversion happens *before* DistributedDataParallel wrapping, as required.

### Why this is useful:
SyncBatchNorm improves performance and stability in multi-GPU distributed training
and is commonly used in ImageNet training recipes. Adding this option makes the
example more complete and aligned with standard training practices.

### Testing:
- Verified that the script runs correctly with dummy data on CPU.
- Ensured that normal execution is unchanged unless `--sync-bn` is passed.

Addresses Issue #793.

### CODE REVIEW

**Comment by snehasonkusare-tech** [comment]
Hi, this is my first contribution to PyTorch. Please let me know if any changes are needed.
Thanks!
